core/src/main/java/org/apache/calcite/rel/rel2sql/PivotRelToSqlUtil.java

```diff
@@ -52,72 +59,89 @@ public class PivotRelToSqlUtil {
    * @return  Result with sqlPivotNode wrap in it.
    */
   public SqlNode buildSqlPivotNode(
-      Aggregate e, SqlImplementor.Builder builder, List<SqlNode> selectColumnList) {
+      Aggregate e, SqlImplementor.Builder builder, List<SqlNode> selectColumnList,
+      List<SqlNode> aggregateColList) {
     //create query parameter
-    SqlNode query = ((SqlSelect) builder.select).getFrom();
+    Optional<RelTrait> pivotRelTrait =
+        e.getTraitSet().stream().filter(relTrait -> relTrait instanceof PivotRelTrait).findAny();
+    boolean checksubquery = ((PivotRelTrait) pivotRelTrait.get()).getsubqueryPresentFlag();
+    String alias = ((PivotRelTrait) pivotRelTrait.get()).getPivotAlias();
+    if (!alias.equals("false")) {
+      pivotTableAlias = alias;
+    }
+    SqlNode query;
+    if (checksubquery) {
+      query = builder.select;
+    } else {
+      query = builder.select.getFrom();
+    }
 
     //create aggList parameter
-    SqlNodeList aggList = getAggSqlNodes(e, selectColumnList);
-
+    SqlNodeList axisNodeList = getAxisNodeList(selectColumnList, checksubquery);
 
     //create axisList parameter
-    SqlNodeList axesNodeList = getAxisSqlNodes(e);
+    SqlNodeList aggList = getAxisSqlNodes(e);
 
     //create inValues List parameter
-    SqlNodeList inColumnList = getInValueNodes(e);
+    SqlNodeList inColumnList = getInValueNodes(selectColumnList, aggregateColList, axisNodeList);
```

Comment:
is this naming, aggregateColList, correct? it seems to store charliterals, correct the name right from the caller of this method as well.

Reply:
i have renamed 'aggreateColumnList'